ninja/operation.py

```diff
@@ -31,6 +31,27 @@
 __all__ = ["Operation", "PathView", "ResponseObject"]
 
 
+def load_request(request: HttpRequest):
+    """
+    Loads a non-json PUT/PATCH request as a POST, reading data from request body
+    and placing them in request.POST and request.FILES.
+    This is required to fix the behaviour when we have a PUT/PATCH endpoint that also
+    receives a File as parameter, forcing the data to be sent as multipart/form-data and
+    making the ParamModel look up at request.POST and request.FILES
+    """
+    if request.method in ("PUT", "PATCH") and request.content_type != "application/json":
+        if hasattr(request, '_post'):
+            del request._post
+            del request._files
+        try:
+            request.method = "POST"
+            request.META['REQUEST_METHOD'] = 'POST'
+            request._load_post_and_files()
+            request.META['REQUEST_METHOD'] = 'PUT'
+            request.method = "PUT"
```

Comment:
Probably best not to convert a PATCH into a PUT needlessly.

```suggestion
    if request.method in ("PUT", "PATCH") and request.content_type != "application/json":
        original_method = request.method
        if hasattr(request, '_post'):
            del request._post
            del request._files
        try:
            request.method = "POST"
            request.META['REQUEST_METHOD'] = 'POST'
            request._load_post_and_files()
            request.META['REQUEST_METHOD'] = original_method
            request.method = original_method
```

Reply:
I got same issue today using form data and PUT method... :(